packages/supplier/README.md

```diff
@@ -0,0 +1,95 @@
+# @truffle/supplier
+
+This package provides infrastructure for the rest of Truffle, defining the
+concept of a "supplier", a component that downloads a specific version of some
+desired code resource according to one or more strategies. (For example,
+@truffle/compile-solidity comprises its "CompilerSupplier", which fetches
+specific solc versions from the web / via docker / et al.)
+
+## Usage example
+
+1. Define a common "results" specification: what does `supplier.load()` and
+   `supplier.list()` return?
+
+   e.g.:
+
+   ```typescript
+   export type Compiler = { compile(): any };
+
+   export namespace Results {
+     export type Specification = {
+       load: Promise<Compiler>;
+       list: Promise<string[]>;
+     };
+   }
+   ```
+
+2. Define one or more strategies.
+
+   e.g.:
+
+   ```typescript
+   import { Mixin } from "ts-mixer";
```

Comment:
Could you make a point of also an including an example of how you'd do things without `ts-mixer`?